### PR TITLE
CORE-1066: Handle "verified_block_hash" in Swift/Java as optional

### PR DESCRIPTION
**Pull request by ewerx**

### CODE REVIEW

File: WalletKitCore/src/crypto/BRCryptoNetwork.c [hunk 1]
```diff
@@ -285,7 +286,9 @@ cryptoNetworkSetHeight (BRCryptoNetwork network,
 
 extern BRCryptoHash
 cryptoNetworkGetVerifiedBlockHash (BRCryptoNetwork network) {
-    return cryptoHashTake (network->verifiedBlockHash);
+    return (NULL != network->verifiedBlockHash)
+    ? cryptoHashTake (network->verifiedBlockHash)
+    : NULL;
```
Comment (EBGToo) [inline WalletKitCore/src/crypto/BRCryptoNetwork.c:291]: All the 'take' functions now accept a `NULL` argument and simply return it.

File: WalletKitJava/crypto/src/main/java/com/breadwallet/crypto/blockchaindb/models/bdb/Blockchain.java [hunk 2]
```diff
@@ -33,7 +33,7 @@ public static Blockchain create(@JsonProperty("id") String id,
                                     @JsonProperty("block_height") UnsignedLong blockHeight,
                                     @JsonProperty("fee_estimates") List<BlockchainFee> feeEstimates,
                                     @JsonProperty("confirmations_until_final") UnsignedInteger confirmationsUntilFinal,
-                                    @JsonProperty("verified_block_hash") String verifiedBlockHash) {
+                                    @JsonProperty("verified_block_hash") Optional<String> verifiedBlockHash) {
```
Comment (EBGToo) [inline WalletKitJava/crypto/src/main/java/com/breadwallet/crypto/blockchaindb/models/bdb/Blockchain.java:36]: Does an `Optional` actually get passed here?  Other classes, see `Block`, annotate the argument as `@Nullable`, save the field as `null` and then the corresponding getter uses `Optional.fromNullable` (see `Block.getRaw()`).  I'd prefer consistency with the existing approach.

**Review by EBGToo — CHANGES_REQUESTED** [review]

File: WalletKitCore/src/crypto/BRCryptoNetwork.c (hunk 1 above)
Comment (ewerx) [reply WalletKitCore/src/crypto/BRCryptoNetwork.c:291]: On refactor branch only?

File: WalletKitJava/crypto/src/main/java/com/breadwallet/crypto/blockchaindb/models/bdb/Blockchain.java (hunk 2 above)
Comment (ewerx) [reply WalletKitJava/crypto/src/main/java/com/breadwallet/crypto/blockchaindb/models/bdb/Blockchain.java]: Fixed

File: WalletKitCore/src/crypto/BRCryptoNetwork.c (hunk 1 above)
Comment (EBGToo) [reply WalletKitCore/src/crypto/BRCryptoNetwork.c:291]: Sorry, right.

**Review by EBGToo — APPROVED** [review]

**Pull request merged**